### PR TITLE
DEVEXP-232: Improved logging and error handling for digest-basic auth

### PR DESCRIPTION
**Pull request by rjrudin**
Added a test for a missing token and for an invalid port. 

Renamed "test-auth-ml" since we now have two authorization plugins.

### CODE REVIEW

File: src/koop/tokenMarkLogicAuthentication.js [hunk 1]
```diff
@@ -17,85 +17,65 @@ function tokenMarkLogicAuthentication(options) {
...
-        resolve(json);
-    } else if (response.authenticated == false) {
-        let err = new Error('Invalid credentials.');
-        err.code = 401;
-        err.httpStatusMessage = response.httpStatusMessage;
-        err.httpStatusCode = response.httpStatusCode;
-        reject(err);
-    }
-    else {
-        let err = new Error("MarkLogic error");
```
Comment (rjrudin) [inline src/koop/tokenMarkLogicAuthentication.js:86]: This code was never reached - the connectAndCacheClient method either returns a "response" object via `resolve` or it tosses an error via `reject`, which is handled by the `catch` method.

**Review by BillFarber — APPROVED** [review]
LGTM

**Comment by rjrudin** [comment]
Just did a rebase off develop, added "start-ssl" as a new script for using the "test-ssl" config.

**Review by BillFarber — APPROVED** [review]
LGTM

**Pull request merged**